src/Control/Effect/Interpret.hs

```diff
@@ -23,27 +77,27 @@ import Control.Monad.Trans.Class
 --   At time of writing, a simple passthrough use of 'runInterpret' to handle a 'State' effect is about five times slower than using 'StateC' directly.
 --
 --   prop> run (runInterpret (\ op -> case op of { Get k -> k a ; Put _ k -> k }) get) === a
-runInterpret :: (forall x . eff m x -> m x) -> InterpretC eff m a -> m a
-runInterpret handler = runReader (Handler handler) . runInterpretC
-
-newtype InterpretC eff m a = InterpretC { runInterpretC :: ReaderC (Handler eff m) m a }
-  deriving (Alternative, Applicative, Functor, Monad, MonadFail, MonadFix, MonadIO, MonadPlus)
+runInterpret
```

Comment:
Could you please document the rank-n-quantification-means-composition-doesn’t-work-like-you’d-expect thing? 🙏

Alternatively, maybe we could export a `newtype` around the `Reifies`-constrained algebra and require callers to pass that in instead, which would be a little more syntactic noise but would allow compositions.

Reply:
I did try to `newtype` the `s` away, but unfortunately it broke all the performance gains. However! The genius @isovector has pointed out that [this patch](https://gitlab.haskell.org/ghc/ghc/merge_requests/668) might actually fix that. I'm going to try now.

Reply:
Alas, even GHC 8.9 seems to still suck :(

```
benchmarked InterpretC vs InterpretStateC vs StateC/InterpretStateC/10000
time                 455.9 μs   (451.7 μs .. 461.2 μs)
                     0.999 R²   (0.998 R² .. 1.000 R²)
mean                 457.9 μs   (455.9 μs .. 460.3 μs)
std dev              7.715 μs   (5.617 μs .. 9.403 μs)

benchmarked InterpretC vs InterpretStateC vs StateC/StateC/10000
time                 22.18 μs   (21.09 μs .. 23.23 μs)
                     0.985 R²   (0.974 R² .. 0.996 R²)
mean                 22.79 μs   (22.17 μs .. 24.93 μs)
std dev              3.487 μs   (1.033 μs .. 7.067 μs)
```

I'll carry on with the approach in this PR for now, can always look at a `newtype` later.

Reply:
Ah well, thanks for looking into it!